internal/controller/api/v1alpha1/installagentenv_types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type InstallAgentEnvSpec struct {
+	// Proxy defines the proxy settings for agents and clusters that use the
+	// InstallationEnvironment. If unset, the agents and clusters will not be 
+	// configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to 
+	// all cluster hosts. They are added to any NTP sources that were configured through
+	// other means.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// SSHAuthorizedKeys is a list of SSH public keys that will be added to all agents 
+	// for use in debugging.
+	// +optional
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// PullSecretRef is the reference to the secret to use when pulling images.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef"`
+
+	// Agents that boot as a result of this InstallationEnvironment should be labeled 
+	// such that they match this selector.
+	// That is how a user would identify which agents are associated with a particular 
+	// InstallationEnvironment.
+	AgentLabelSelector metav1.LabelSelector `json:"agentLabelSelector"`
+
+	// Any labels to apply to agents at creation time that are associated with this 
+	// InstallationEnvironment.
+	// +optional
+	AgentLabels map[string]string `json:"agentLabels,omitempty"`
+
+	// ClusterDeploymentRef is the reference to the single ClusterDeployment that will 
+	// be installed from this InstallationEnvironment.
+	// Future versions will allow for multiple ClusterDeployments and this reference 
+	// will be removed.
+	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef,omitempty"`
```

Comment:
Cluster is one that need to use it , does it make sence to move it to cluster deployment maybe under agent install strategy?
If it's here then every time i need to reconcile cluster i need to list all `InstallAgentEnvSpec` and search for the right one for me. 
@dgoodwin @mhrivnak what do you think?

Reply:
currently AdditionalNtpSource, Proxy configuration and ssh public key, maybe we will change it in the future but because we are in alpha i suggest to put it in cluster deployment.

Reply:
Can be resolved, currently we are going to search for the right reference on every reconcile.